tests/test_misc.rs

```diff
@@ -80,3 +80,115 @@ fn test_big_endian() {
     // so we can see it in the CI logs
     assert_eq!(u16::to_be_bytes(1), [0, 1]);
 }
+
+#[test]
+fn test_new_encoder() {
+    let arr3 = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]; // 2 * 2 * 3
+    let arr4 = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]; // 2 * 2 * 4
+
+    let enc = qoi::Encoder::new(&arr3, 2, 2).unwrap();
```

Comment:
Don't need all these unwraps if you make the test return `Result<()>`

Reply:
ack